demos/yielding/README.md

```diff
@@ -0,0 +1,47 @@
+# Yielding
+
+In this demo we show the usage of yielding, which occurs in a computer program during multithreading.
+
+By calling syscall `yield`, the thread gives away the utilization of the CPU to another thread, or a specific thread (discussed in another demo).
+
+To describe thread yielding, we need to introduce:
+
+* [`add_thread_from_worker`](https://github.com/DavidSpickett/ARMMultiTasking/blob/master/src/user/thread.c#L42)
+* [`do_scheduler`](https://github.com/DavidSpickett/ARMMultiTasking/blob/master/src/kernel/thread.c#L353)
+* [`yield`](https://github.com/DavidSpickett/ARMMultiTasking/blob/master/src/user/thread.c#L92)
+
+## [`add_thread_from_worker`](https://github.com/DavidSpickett/ARMMultiTasking/blob/master/src/user/thread.c#L42)
+
+In AMT we store all of the threads in an array called `all_threads`.
```

Comment:
I would move/merge this line into the `do_scheduler` description. That's where it's most relevant. Here it's a bit of a non sequitur.

(roughly)
AMT stores threads in all_threads. The scheduler walks this array .. etc etc.